core/integration/directory_test.go

```diff
@@ -521,6 +521,13 @@ func TestDirectoryWithoutDirectoryWithoutFile(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, []string{"some-dir", "some-file"}, entries)
 
+	// Test without dir with allow not found = false
+	_, err = dir1.
+		WithoutDirectory("non-existant", dagger.DirectoryWithoutDirectoryOpts{AllowNotFound: false}).
```

Comment:
@TomChv this is your issue.

In go, the empty struct defaults to `false` - therefore, we have the API constraint today that all boolean options must default to false.

What you need is to invert the option, to something like `MustExist: true`, so that the default is `false`.